packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/ConfirmUnverifiedModal.tsx

```diff
@@ -53,47 +34,50 @@ export const ConfirmUnverifiedModal = ({
         ? 'TR_PLEASE_ENABLE_PASSPHRASE'
         : 'TR_PLEASE_CONNECT_YOUR_DEVICE';
 
-    const enablePassphraseAndContinue = async () => {
-        if (!device.available) {
-            const result = await dispatch(applySettings({ use_passphrase: true }));
-            if (!result || !result.success) return;
+    const handleClose = () => dispatch(onCancel());
+    const handleEvent = () => {
+        action.event();
+
+        if (action.closeAfterEventTriggered) {
+            handleClose();
         }
-        dispatch(verify());
     };
-    const continueUnverified = () => dispatch(showUnverified());
-    const close = () => dispatch(onCancel());
 
     return (
-        <StyledModal
-            heading={<Translation id={deviceStatus} values={{ deviceLabel: device.label }} />}
-            isCancelable
-            onCancel={close}
-            description={
-                <Translation
-                    id={warningText}
-                    values={{ claim: <Translation id={description} /> }}
-                />
-            }
-            bottomBarComponents={
+        <NewModal
+            variant="warning"
+            size="small"
+            icon="shieldWarning"
+            onCancel={handleClose}
+            bottomContent={
                 <>
-                    <Button variant="warning" onClick={continueUnverified}>
-                        <Translation id={showUnverifiedButtonText} />
+                    <Button variant="warning" onClick={handleEvent}>
+                        <Translation id={action.title} />
                     </Button>
-
                     {isPassphraseRequired && (
-                        <StyledButton
+                        <Button
                             variant="primary"
                             onClick={enablePassphraseAndContinue}
                             isDisabled={isDeviceLocked}
                         >
                             <Translation id="TR_ACCOUNT_ENABLE_PASSPHRASE" />
-                        </StyledButton>
+                        </Button>
                     )}
-                    <Button onClick={close} variant="tertiary">
+                    <Button onClick={handleClose} variant="tertiary">
                         <Translation id="TR_BACK" />
                     </Button>
                 </>
             }
-        />
+        >
+            <H3>
+                <Translation id={deviceStatus} values={{ deviceLabel: device.label }} />
+            </H3>
```

Comment:
Could you use the `heading` prop?

Reply:
The prop `heading` will add the title to the modal's header. And it doesn't look good in order header -> icon -> paragraph.

Reply:
What does @adamhavel think?